integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -1276,6 +1276,20 @@ def test_generic_reductions(data_gen):
             'count(1)'),
         conf=local_conf)
 
+@ignore_order(local=True)
+def test_hash_groupby_with_minby():
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: three_col_df(spark, int_gen, int_gen, UniqueLongGen())
+            .groupby('a').agg(f.min_by('b', 'c'))
+    )
+
+@ignore_order(local=True)
+def test_reduction_with_minby():
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark: two_col_df(spark, int_gen, UniqueLongGen()).selectExpr(
+            "min_by(a, b)")
+    )
+
```

Comment:
Please add test cases for Float type (has Nan and without Nan).

Reply:
Will do it once the JNI jar is ready.

Reply:
Could we expand the testing to not just compare longs, but other types too. It shows that you support all kinds of types for the ordering column, and it would be nice to verify that it does actually work.  I know it is a little tricky with the ambiguity in the generated data.